### PR TITLE
re #1198 work around +size suffixes in CAPI responses

### PR DESCRIPTION
**Pull request by inorton**
Strip the extra + chars in surface port names that are wrongly returned by the CAPI.

Also rely on station ID checks rather than name comparison before we assume the CAPI is lagging

### CODE REVIEW

File: companion.py [hunk 1]
```diff
@@ -670,6 +670,11 @@ def station(self) -> CAPIData:  # noqa: CCR001
             logger.warning("No lastStarport name!")
             return data
 
+        # work around issue 1198 -
+        # https://issues.frontierstore.net/issue-detail/40706
+        # strip "+" chars off star port names returned by the CAPI
```
Comment (A-UNDERSCORE-D) [inline companion.py:675]:
```suggestion
        # WORKAROUND: n/a | 06-08-2021: Issue 1198 and https://issues.frontierstore.net/issue-detail/40706 -- strip "+" chars off star port names returned by the CAPI
```

File: companion.py [hunk 2]
```diff
@@ -687,22 +692,22 @@ def station(self) -> CAPIData:  # noqa: CCR001
...
-            if last_starport_name != shipdata['name'] or last_starport_id != int(shipdata['id']):
-                logger.warning(f"{last_starport_name!r} != {shipdata['name']!r} or "
-                               f"{last_starport_id!r} != {int(shipdata['id'])!r}")
+            if last_starport_id != int(shipdata['id']):
+                logger.warning(f"{last_starport_id!r} != {int(shipdata['id'])!r}")
                 raise ServerLagging()
 
             else:
+                shipdata['name'] = last_starport_name
                 data['lastStarport'].update(shipdata)
```
Comment (A-UNDERSCORE-D) [inline companion.py:712]:
```suggestion
# WORKAROUND END

```

**Review by A-UNDERSCORE-D — CHANGES_REQUESTED** [review]

**Comment by A-UNDERSCORE-D** [comment]
Please switch this to point at `develop`, other than the requested comment changes to fit with [the new standard](https://github.com/EDCD/EDMarketConnector/blob/main/Contributing.md#comments) this LGTM

**Review by A-UNDERSCORE-D — APPROVED** [review]

**Comment by A-UNDERSCORE-D** [comment]
Looks good, should work fine but Ill let @Athanasius give it a look over in case I missed any gotchas

**Comment by Athanasius** [comment]
> Looks good, should work fine but Ill let @Athanasius give it a look over in case I missed any gotchas

That all looks good, I'll give it a test this afternoon and then merge if I don't run into any issues.

**Pull request merged**